docs/vocabulary.md

```diff
@@ -388,6 +389,230 @@ pipeline in corresponding documentation.
 | **Tekton**         | N/A       | Step       | Task     | Pipeline | Trigger | Resource (?)           |
 | **Zuul**           | N/A       | N/A        | Job      | Pipeline | Trigger | Node (?)               |
 
+### Pipeline Step Types
+
+The table below is an attempt to create a mapping of common names for the various types of Pipeline Steps (the unit of work two degrees smaller than a Pipeline). (The effort of proposing a common vocabulary across CI/CD Tools and Technologies is not yet performed, but we'll use the most common term from the table above as the working term.)
+
+While CI/CD tools and technologies generally give developers broad leeway in naming and implementing their Pipeline Steps, agreeing upon some common terms will help develop more pluggable pipelines. This list is not meant to be exhaustive.
+
+Each type of Pipeline Step will generally have several implementations that are tool specific. For example, Source will have implementations for various Source Code Management (SCM) tools. Publish will have implementations for each type of repository, and so on.
+
+#### Setup
+* Semantics: Provision build resources, set up the build workspace.
+* Aliases: Initialize, Start, Prepare, Workspace
+* Inputs: Build Request Parameters, Build Container Image Name and Version
+* Outputs: Secrets, Build Environment, Build Workspace
+* Other Results and Side Effects: N/A
+
+#### Source
+* Semantics: Download, retrieve or copy software, images, and documentation into the build workspace. Fetch configuration data.
+* Aliases: Clone, Fetch
+* Inputs: Source Code Reference (Repository, Branch, Commit)
+* Outputs: Software Source, Binary Source
+* Other Results and Side Effects: N/A
+
+#### Tag
+* Semantics: Annotate source code, images, etc. with information, such as the version number and a description.
+* Aliases: Annotate, Version
+* Inputs: Source Code Reference (Repository, Branch, Commit), Software Source, Binary Source, Generated Software, Generated Binaries
+* Outputs: Tag/annotation metadata
+* Other Results and Side Effects: Tags/annotations have been added to the software.
+
+#### Secret Detection
```

Comment:
This is definitely a grey area! It could be a type of scan, but the inputs are a bit different. 

When I've seen secret detection implemented, it's often coupled with, or even run by, the source code management server.  So, the source branch information is important, but the generated artifacts (post-build) are not.  In fact, since the point of secret detection is to keep things out of SCM, you generally want to exclude generated artifacts from this scan.  Even with pre-commit hooks, the SCM information is on hand.

Most scanning tools I've seen are SCM-agnostic, so they assume you've already pulled your code from SCM and done a build if you need it.  They ignore the source branch information, but they might need the generated artifacts, depending on the tool and how you configure it.

We could come up with counter-examples to both, I'm sure! 

If it's OK with you we could merge in the initial version with secret detection as its own category, and then open an issue to debate this part in particular.

Reply:
We could merge it and create a issue to debate it. The important part would be do describe for a reader, not having seen the discussion, why secret detection would be its own step.

Reply:
Opened an issue: https://github.com/cdfoundation/sig-interoperability/issues/93